packages/flutter_tools/lib/runner.dart

```diff
@@ -155,6 +193,48 @@ Future<int> _handleToolError(
   }
 }
 
+String _crashCommand(List<String> args) => 'flutter ${args.join(' ')}';
+
+String _projectMetadataInformation() {
+  FlutterProject project;
+  try {
+    project = FlutterProject.current();
+  } catch (_) {
```

Comment:
this should probably `catch Exception`, to avoid catching NoSuchMethodError, TypeError, and friends. It would probably be something like a FormatException or a FileSystemException, right?

Reply:
Looking through the code I could only see it throwing when the pubspec is malformed, which come to think of it, would be very interesting in an issue report.  Will make this return the exception text